src/location.rs

```diff
@@ -1,10 +1,43 @@
 use bevy::prelude::*;
+use crate::direction::{Direction};
 
-pub type XCoord = u8;
-pub type YCoord = u8;
+type Coord = u8;
 
-#[derive(Component, Default, Debug)]
+const MAX_X: Coord = 15;
+const MAX_Y: Coord = 8;
+
+#[derive(Component, Debug)]
 pub struct Location {
-    x: XCoord,
-    y: YCoord,
+    x: Coord,
```

Comment:
I quite liked the separate X and Y types so you can't assign the wrong dimension.